presto-main/src/main/java/io/prestosql/dispatcher/LocalDispatchQueryFactory.java

```diff
@@ -109,6 +110,14 @@ public DispatchQuery createDispatchQuery(
                 warningCollector,
                 StatementUtils.getQueryType(preparedQuery.getStatement().getClass()));
 
+        // It is important that `queryCreatedEvent` is called here. Moving it past the `executor.submit` below
+        // can result in delivering query-completed event before query-created event.
```

Comment:
That's not exactly true, is it? Actually, i think before this PR, there seemed to be no possibility for out of order on create/complete events.

I think the more important bit is that this must be here so that query-created is fired before analysis, which may invoke plugins.

Reply:
> I think the more important bit is that this must be here so that query-created is fired before analysis, which may invoke plugins.

:+1 

And also query-completed should be called after any interaction with plugins.

Reply:
> That's not exactly true, is it? Actually, i think before this PR, there seemed to be no possibility for out of order on create/complete events

Yeah. It was not possible; because of `queryMonitor` nature explained before. I will change the first part of comment

Reply:
> And also query-completed should be called after any interaction with plugins

This call spot has nothing to do with that

Reply:
@findepi @kokosing do you like the phasing more?